great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -840,23 +841,17 @@ def execution_engine_type(self) -> Type[SqlAlchemyExecutionEngine]:
 
     def get_engine(self) -> sqlalchemy.engine.Engine:
         if self.connection_string != self._cached_connection_string or not self._engine:
-            # validate that SQL Alchemy was successfully imported and attempt to create an engine
-            if SQLALCHEMY_IMPORTED:
-                try:
-                    self._engine = sqlalchemy.create_engine(self.connection_string)
-                except Exception as e:
-                    # connection_string has passed pydantic validation, but still fails to create a sqlalchemy engine
-                    # one possible case is a missing plugin (e.g. psycopg2)
-                    raise SQLDatasourceError(
-                        "Unable to create a SQLAlchemy engine from "
-                        f"connection_string: {self.connection_string} due to the "
-                        f"following exception: {str(e)}"
-                    ) from e
-                self._cached_connection_string = self.connection_string
-            else:
+            try:
```

Comment:
@billdirks If we indeed dispense with the `<MY_PACKAGE>_IMPORTED` paradigm and this pull request gets merged, then how should we go about updating all the other `Datasource` subclasses?  Perhaps create a ticket for each?  As it stands right now, all `Datasource` implementations use this `<MY_PACKAGE>_IMPORTED` pattern (I followed the recommendation by @NathanFarmer and made all File Path type `Datasource` modules follow it).  Thank you.

Reply:
I think we should do it across the whole code base. We set `module = None` all over the place. We can fix it as we go or do it all at once. I'm more worried about the cases where we don't catch it at all than catching it an alternative way.

Reply:
@billdirks We have a lot less of `module = None` now, because of the effort already conducted (the one mentioned in my comment).  But whatever is our best approach going forward, I am happy for us to apply universally.  Thanks.